opm/simulators/flow/Main.hpp

```diff
@@ -639,10 +640,18 @@ class Main
             return EXIT_FAILURE;
         }
 
-        if (phases.size() == 3) { // oil water brine case
-            flowEbosOilWaterBrineSetDeck(
-                setupTime_, deck_, eclipseState_, schedule_, summaryConfig_);
-            return flowEbosOilWaterBrineMain(argc_, argv_, outputCout_, outputFiles_);
+        if (phases.size() == 3) { 
+
+            if (phases.active(Phase::OIL)){ // oil water brine case
+                flowEbosOilWaterBrineSetDeck(
+                    setupTime_, deck_, eclipseState_, schedule_, summaryConfig_);
+                return flowEbosOilWaterBrineMain(argc_, argv_, outputCout_, outputFiles_);
+            }
+            if (phases.active(Phase::GAS)){ // gas water brine case
+                flowEbosGasWaterBrineSetDeck(
+                    setupTime_, deck_, eclipseState_, schedule_, summaryConfig_);
+                return flowEbosGasWaterBrineMain(argc_, argv_, outputCout_, outputFiles_);
+            }
```

Comment:
There should be
```C++
return EXIT_FAILURE
```
possibly accompanied by some logging, here to cover the **unlikely** case that neither oil nor gas are active.